.github/workflows/create-release-candidate.yml

```
@@ -237,29 +228,58 @@ jobs:
           shasum cocoapods/embrace_*.zip
           aws s3 cp --acl=public-read cocoapods/embrace_*.zip ${{ env.AWS_BUCKET_STAGING }}
           rm -rf cocoapods
-
           # missing dsysm upload
-
       - name: Flush CloudFront CDN
         env:
           # downloads.stg.emb-eng.com
           AWS_CLOUDFRONT_ID: E1Q329634TJ1Q9
         run: |
           echo "$(date): Invalidating https://us-east-1.console.aws.amazon.com/cloudfront/v3/home?region=us-west-2#/distributions/${{ env.AWS_CLOUDFRONT_ID }}"
           aws cloudfront create-invalidation --region us-west-2 --distribution-id ${{ env.AWS_CLOUDFRONT_ID }} --paths '/*' | tee invalidation.json
-
           invalidation_id=$(cat invalidation.json | jq -r .Invalidation.Id)
-
           echo "$(date): Waiting for completion of https://us-east-1.console.aws.amazon.com/cloudfront/v3/home?region=us-west-2#/distributions/${{ env.AWS_CLOUDFRONT_ID }}/invalidations/details/${invalidation_id}"
           aws cloudfront wait invalidation-completed --region us-west-2 --distribution-id ${{ env.AWS_CLOUDFRONT_ID }} --id $invalidation_id
 
+  create_github_release:
+    name: Create GitHub Release
+    runs-on: ubuntu-latest
+    needs:
+      - rc_version_number
+      - archive_cocoapods_artifacts
+
+    steps:
+      - name: Download Cocoapods Release Archive
+        uses: actions/download-artifact@v4
+        with:
+          name: Cocoapods Release Archive
+          path: cocoapods/
+
+      - name: Disable GitHub CLI Prompt if Enabled
+        run: |
+          if [ "$(gh config get prompt)" = "enabled" ]; then
+            gh config set prompt disabled
+          fi
+
+      - name: Create/Edit (Pre) Release
+        env:
+          GITHUB_TOKEN: ${{ github.token }}
+          RC_VERSION: ${{ needs.rc_version_number.outputs.rc_version}}
+        run: |
+          if gh release view --repo embrace-io/embrace-apple-sdk $RC_VERSION > /dev/null 2>&1; then
+            echo "Release $RC_VERSION already exists; editing..."
+            gh release upload $RC_VERSION cocoapods/embrace_$RC_VERSION.zip --clobber --repo embrace-io/embrace-apple-sdk
+          else 
+            echo "Creating Release $RC_VERSION in Github"
+            gh release create $RC_VERSION cocoapods/embrace_$RC_VERSION.zip --title "Release $RC_VERSION" --prerelease --repo embrace-io/embrace-apple-sdk
```

nitpick - use just the actual version as release name, it looks slightly nicer in github UI:
```suggestion
            gh release create $RC_VERSION cocoapods/embrace_$RC_VERSION.zip --title "$RC_VERSION" --prerelease --repo embrace-io/embrace-apple-sdk
```